tests/unit/nodeManager.spec.js

```diff
@@ -0,0 +1,143 @@
+import { addIdToNodeAndSetUpDiagramReference, addNodeToProcess, getTargetProcess } from '@/components/nodeManager';
```

Comment:
I really like that you've done the code extraction and added unit tests! Great job. 💯